lldb/packages/Python/lldbsuite/test/tools/lldb-dap/lldbdap_testcase.py

```diff
@@ -335,6 +335,37 @@ def cleanup():
                 response["success"], "attach failed (%s)" % (response["message"])
             )
 
+    def attach_by_port(
```

Comment:
could you just extend the `def attach` function? Having a totally new entry point just adds more maintainance burden when doing refactors.